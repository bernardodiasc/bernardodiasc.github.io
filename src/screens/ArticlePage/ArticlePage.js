import React, { Component } from 'react'

import Meta from 'components/Meta'
import LayoutContainer from 'containers/LayoutContainer'
import PageContent from 'displays/PageContent'
import TextBlock from 'displays/TextBlock'
import MarkdownRenderer from 'displays/MarkdownRenderer'

class ArticlePage extends Component {
  static defaultProps = {
    article: {
      title: '',
      excerpt: '',
      body: '',
    },
  }

  render() {
    const { title, excerpt, body } = this.props.article
    return (
      <LayoutContainer>
        <Meta
          title={title}
          description={excerpt}
        />
        <PageContent>
          <TextBlock>
            <MarkdownRenderer text={body} />
          </TextBlock>
        </PageContent>
      </LayoutContainer>
    )
  }
}

export default ArticlePage
