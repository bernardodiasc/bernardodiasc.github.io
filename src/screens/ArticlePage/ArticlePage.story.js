import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs/react'
import StoryRouter from 'storybook-router'
import Provider from 'stories/containers/Provider'
import Docs from 'stories/containers/Docs'

import ArticlePage from './ArticlePage'
import ArticlePageReadme from './README.md'
storiesOf('Screens/ArticlePage', module)
  .addDecorator(story => <Provider story={story()} />)
  .addDecorator(StoryRouter())
  .addDecorator(withKnobs)
  .add('Info', () => <Docs>{ArticlePageReadme}</Docs>)
  .add('Example', () => <ArticlePage />)
  .add('Minimal render', () => <ArticlePage />)
