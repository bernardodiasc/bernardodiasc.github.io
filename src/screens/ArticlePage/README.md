# ArticlePage

## How to use

```
import ArticlePage from 'screens/ArticlePage'
```

```
<ArticlePage />
```

## Properties

| propName | propType | defaultValue | isRequired |
| - | - | - | - |
