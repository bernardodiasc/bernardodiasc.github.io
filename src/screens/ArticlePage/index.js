import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import { getAllArticleDetails } from 'selectors/data'

import ArticlePage from './ArticlePage'

const mapStateToProps = (store, props) => ({
  article: getAllArticleDetails(store, props.article),
})

const mapDispatchToProps = dispatch => (
  bindActionCreators({
  }, dispatch)
)

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(ArticlePage)
