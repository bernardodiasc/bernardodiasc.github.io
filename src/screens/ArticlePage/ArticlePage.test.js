import React from 'react'
import { shallow } from 'enzyme'

import ArticlePage from './ArticlePage.js'

it('renders without crashing', () => {
  const component = shallow(<ArticlePage />)
  expect(component).toMatchSnapshot()
})
