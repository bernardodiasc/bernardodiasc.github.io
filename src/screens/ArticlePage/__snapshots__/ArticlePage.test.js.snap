// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Connect(LayoutContainer)>
  <Meta
    description=""
    title=""
  />
  <PageContent>
    <TextBlock>
      <MarkdownRenderer
        text=""
      />
    </TextBlock>
  </PageContent>
</Connect(LayoutContainer)>
`;
